api-lib/circleAdmin.ts

```diff
@@ -23,7 +23,12 @@ const middleware =
         session_variables: sessionVariables,
       } = composeHasuraActionRequestBody(circleIdInput).parse(req.body);
 
-      if (sessionVariables.hasuraRole !== 'admin') {
+      if (sessionVariables.hasuraRole === 'admin') {
+        await handler(req, res);
+        return;
+      }
+
+      if (sessionVariables.hasuraRole === 'user') {
```

Comment:
Given the new pattern you're implementing, this block requires a handler invocation at the end of its scope. This is also the default case.

Reply:
whoops, good catch